src/dataflow/v2/controlflow-version-two.ts

```diff
@@ -0,0 +1,87 @@
+import { Expression } from 'xpath-ts2'
+import { Environment } from '../v1'
+
+export enum ExecutionState{
+	Always,
+	Maybe,
+	Never
+}
+
+
+export interface IExecutionTuple {
+	readonly executed:               ExecutionState,
+	readonly influencingExpressions: Expression[] //TODO: correct?
+}
+
+export class ExecutionTuple implements IExecutionTuple{
+	readonly executed:               ExecutionState
+	readonly influencingExpressions: Expression[]
```

Comment:
On that note: auch arrays kann man immutable/`readonly` machen :) sind ja ebenfalls referenztypen :)